rllib/algorithms/iql/iql.py

```diff
@@ -0,0 +1,202 @@
+from ray.rllib.algorithms.marwil.marwil import MARWIL, MARWILConfig
+from ray.rllib.connectors.common.add_observations_from_episodes_to_batch import (
+    AddObservationsFromEpisodesToBatch,
+)
+from ray.rllib.connectors.learner.add_next_observations_from_episodes_to_train_batch import (  # noqa
+    AddNextObservationsFromEpisodesToTrainBatch,
+)
+from ray.rllib.core.learner.learner import Learner
+from ray.rllib.core.rl_module.rl_module import RLModuleSpec
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.typing import LearningRateOrSchedule, RLModuleSpecType
+
+
+class IQLConfig(MARWILConfig):
+    """Defines a configuration class from which a new IQL Algorithm can be built
+
+    .. testcode::
+        :skipif: True
+
+        from ray.rllib.algorithms.iql import IQLConfig
+        # Run this from the ray directory root.
+        config = IQLConfig().training(lr=0.00001, gamma=0.99)
+        config = config.offline_data(
+            input_="./rllib/tests/data/pendulum/pendulum-v1_enormous")
+
+        # Build an Algorithm object from the config and run 1 training iteration.
+        algo = config.build()
+        algo.train()
+
+    .. testcode::
+        :skipif: True
+
+        from ray.rllib.algorithms.iql import IQLConfig
+        from ray import tune
+        config = IQLConfig()
+        # Print out some default values.
+        print(config.beta)
+        # Update the config object.
+        config.training(
+            lr=tune.grid_search([0.001, 0.0001]), beta=0.75
+        )
+        # Set the config object's data path.
+        # Run this from the ray directory root.
+        config.offline_data(
+            input_="./rllib/tests/data/pendulum-v1_enormous"
+        )
+        # Set the config object's env, used for evaluation.
+        config.environment(env="Pendulum-v1")
+        # Use to_dict() to get the old-style python config dict
+        # when running with tune.
+        tune.Tuner(
+            "IQL",
+            param_space=config.to_dict(),
+        ).fit()
+    """
+
+    def __init__(self, algo_class=None):
+        super().__init__(algo_class=algo_class or IQL)
+
+        # fmt: off
+        # __sphinx_doc_begin__
+        # The temperature for the actor loss.
+        self.beta = 0.1
+
+        # The expectile to use in expectile regression.
+        self.expectile = 0.8
+
+        # The learning rates for the actor, critic and value network(s).
+        self.actor_lr = 3e-4
+        self.critic_lr = 3e-4
+        self.value_lr = 3e-4
+        # Set `lr` parameter to `None` and ensure it is not used.
+        self.lr = None
+
+        # If a twin-Q architecture should be used (advisable).
+        self.twin_q = True
+
+        # How often the target network should be updated.
+        self.target_network_update_freq = 0
+        # The weight for Polyak averaging.
+        self.tau = 1.0
+
+        # __sphinx_doc_end__
+        # fmt: on
+
+    @override(AlgorithmConfig)
```

Comment:
We subclass from `MARWILConfig`. -> `override(MARWILConfig)`

Reply:
Great catch!! And done :)